hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/api/Matchers.kt

```diff
@@ -0,0 +1,61 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api
+
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.AttributeSearchException
+import java.time.LocalDate
+import java.time.LocalDateTime
+import kotlin.reflect.KClass
+
+data class Matchers(
+  val joinType: JoinType,
+  val stringMatchers: List<StringMatcher>? = null,
+  val booleanMatchers: List<BooleanMatcher>? = null,
+  val integerMatchers: List<IntegerMatcher>? = null,
+  val dateMatchers: List<DateMatcher>? = null,
+  val dateTimeMatchers: List<DateTimeMatcher>? = null,
```

Comment:
I'd quite like to move all of the type matchers into a single `List<TypeMatcher>` but I'll have to do something clever in Jackson to let it work out which type of TypeMatcher to create. Might give this a try but timebox it.